### PR TITLE
Added components for new login page layout

### LINKED ISSUE #98: Make changes for the New Login Page
## Is your feature request related to a problem? Please describe

After discussing with the UI team we received design for the new Sign-up page. Need to reflect the same in the code.

## Describe the solution you'd like

- Make sure the page is responsive.
- User should be able to login using Providers.
- User should be able to login using Credentials.
- Make sure to make reusable components.
- Add Localization for the page.

### PR DESCRIPTION
**Pull request by picksitquick**
Fixes #98 

## Proposed Changes

  - Updated Sign-in page with new layout.
  - Made the page responsive for different screens.
  - Added new reusable components for Text-Box, Socials etc.
  - Added SVGs for login page icons.
  - Changes have been tested locally.

## Screenshots:
Screenshot 2024-03-12 010135

### CODE REVIEW

File: src/app/(auth)/components/LoginBgImage.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+export default function LoginBgImage() {
+    return (
+      <div
+        className="hidden bg-cover lg:block lg:w-3/5 h-screen"
+        style={{
+          backgroundImage: `url("https://cdn.leonardo.ai/users/58b4ebe8-31b7-423d-8587-b03acce30173/generations/5d8ef9a3-f1fa-4373-832b-76038240b7b8/DreamShaper_v7_a_heros_journey_1.jpg")`,
```
Comment (hybridx) [inline src/app/(auth)/components/LoginBgImage.tsx:7]: If possible we can just download this image and add it to our project (for future)

File: src/app/(auth)/components/SocialButton.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  socialName: string;
+  onClick: () => void;
+}
+
+export default function SocialButton({ color, iconPath, socialName, onClick }: SocialButtonProps) {
+  return (
+    <Button
+      variant="contained"
+      style={{ borderRadius: '24px', backgroundColor: color,}}
+      className='font-medium font-roboto h-12 w-36'
```
Comment (hybridx) [inline src/app/(auth)/components/SocialButton.tsx:17]: I think you can remove this and use a single font across all the components and pages

File: src/app/(auth)/components/TextBox.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
+}
+
+export default function TextBox({id, height,placeholder, label, type, onChange}:TextBoxProps){
+    return(
+        <div className={`relative w-full`}>
+            <input onChange={onChange} type={type} id={id} 
+                className={`block px-2.5 pb-2.5 pt-4 w-full h-${height} text-md text-black rounded-md border border-gray-400 appearance-none focus:outline-none focus:ring-0 focus:border-orange-500 focus:border-2 peer hover:border-black focus:placeholder-gray-400 placeholder-transparent`} placeholder={placeholder} />
+            <label htmlFor={id} 
+                className="absolute text-md text-gray-500  duration-150 transform -translate-y-4 scale-75 top-2 z-10 origin-[0] bg-white px-2 peer-focus:px-2 peer-focus:text-orange-500 peer-placeholder-shown:scale-100 peer-placeholder-shown:-translate-y-1/2 peer-placeholder-shown:top-1/2 peer-focus:top-2 peer-focus:scale-75 peer-focus:-translate-y-4 rtl:peer-focus:translate-x-1/4 rtl:peer-focus:left-auto start-3 peer-focus:font-semibold">{label}</label>
```
Comment (hybridx) [inline src/app/(auth)/components/TextBox.tsx:18]: We don't have a mui component for this?

**Review by hybridx — CHANGES_REQUESTED** [review]
Minor changes, rest LGTM

File: src/app/(auth)/components/TextBox.tsx (hunk 3 above)
Comment (picksitquick) [reply src/app/(auth)/components/TextBox.tsx:18]: We do but that isn't flexible to use beyond a certain extent, I have an MUI component named "TextField".
Will make the necessary changes.

**Review by hybridx — APPROVED** [review]

**Pull request merged**